### ISSUE TITLE
🐛 Message: byte index 1 is not a char boundary

### ISSUE THREAD

**Issue by YangFong** [biomejs/biome#4253]
### Environment information

```block
CLI:
  Version:                      1.9.3
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.15.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.8.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Log:

```sh
 pnpm biome check --write .
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_10
Message: byte index 1 is not a char boundary; it is inside '丽' (bytes 0..3) of `丽萨`

Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_factory/src/utils.rs:34:36
Thread Name: biome::worker_6
Message: byte index 1 is not a char boundary; it is inside '商' (bytes 0..3) of `商品名称`
```

See the code: [playground](https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAZABhAHQAYQBTAG8AdQByAGMAZQA6ACAAYQBuAHkAWwBdACAAPQAgAFsAXQAKAAoAZgBvAHIAIAAoAGwAZQB0ACAAaQAgAD0AIAAwADsAIABpACAAPAAgADUAMAA7ACAAaQArACsAKQAgAHsACgAgACAAZABhAHQAYQBTAG8AdQByAGMAZQAuAHUAbgBzAGgAaQBmAHQAKAB7AAoAIAAgACAAIABpAGQAOgAgAGkALAAKACAAIAAgACAAdABpAHQAbABlADoAIAAnAD1OKIQnACAAKwAgAGkALAAKACAAIAAgACAAYQB2AGEAdABhAHIAOgAgACcALwBpAG0AYQBnAGUAcwAvAHUAcwBlAHIALgBqAHAAZwAnACwACgAgACAAIAAgAGMAbwBuAHQAZQBuAHQAOgAgAFsACgAgACAAIAAgACAAIAB7AAoAIAAgACAAIAAgACAAIAAgAGwAYQBiAGUAbAA6ACAAJwBPAHcAbgBlAHIAJwAsAAoAIAAgACAAIAAgACAAIAAgAHYAYQBsAHUAZQA6ACAAMgA5ADAAMwAgACsAIABpACwACgAgACAAIAAgACAAIAB9ACwACgAgACAAIAAgACAAIAB7AAoAIAAgACAAIAAgACAAIAAgAGwAYQBiAGUAbAA6ACAAJwAAX8tZ9mX0lScALAAKACAAIAAgACAAIAAgACAAIAB2AGEAbAB1AGUAOgAgAG4AZQB3ACAARABhAHQAZQAoACkALgB0AG8ATABvAGMAYQBsAGUAUwB0AHIAaQBuAGcAKAApACwACgAgACAAIAAgACAAIAB9ACwACgAgACAAIAAgACAAIAB7AAoAIAAgACAAIAAgACAAIAAgAGwAYQBiAGUAbAA6ACAAJwDbj6ZeJwAsAAoAIAAgACAAIAAgACAAIAAgAHYAYQBsAHUAZQA6ACAAaQAsAAoAIAAgACAAIAAgACAAfQAsAAoAIAAgACAAIABdACwACgAgACAAfQApAAoAfQAKAAoAZQB4AHAAbwByAHQAIAB7ACAAZABhAHQAYQBTAG8AdQByAGMAZQAgAH0ACgA%3D)

### Expected result

it should not throw an error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Duplicate of https://github.com/biomejs/biome/issues/4181.
This will be fixed in the next patch release.
